### PR TITLE
Add aten::mode

### PR DESCRIPTION
**Pull request by hjhee**
- mode
- mode.values

### CODE REVIEW

File: yaml/xpu_functions.yaml [hunk 1]
```diff
@@ -722,3 +722,5 @@ supported:
   - _weight_norm_interface
   - _weight_norm_interface_backward
   - range.out
+  - mode
+  - mode.values
```
Comment (majing921201) [inline yaml/xpu_functions.yaml:726]: no need to register, CUDA also doesn't do

File: src/ATen/native/xpu/sycl/TensorModeKernel.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include <cstdint>
+
+namespace at {
+class Tensor;
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/TensorModeKernel.h:5]: Why we need this here ?

File: src/ATen/native/xpu/sycl/TensorModeKernel.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include <cstdint>
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/TensorModeKernel.h:2]: Is the head file needed ?

File: src/ATen/native/xpu/sycl/TensorModeKernel.cpp [hunk 4]
```diff
@@ -0,0 +1,834 @@
...
+#include <ATen/Dispatch.h>
+#include <ATen/native/ReduceOpsUtils.h>
+#include <ATen/native/Resize.h>
+#include <comm/SYCLContext.h>
+#include <comm/SYCLHelpers.h>
+#include <comm/TensorInfo.h>
+
+namespace at::native::xpu {
+
+namespace {
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/TensorModeKernel.cpp:11]: Can we remove this ambiguous namespace ?

File: src/ATen/native/xpu/sycl/TensorCompareKernels.cpp [hunk 5]
```diff
@@ -1,10 +1,11 @@
 #include <ATen/ATen.h>
 #include <ATen/Dispatch.h>
 #include <ATen/NumericUtils.h>
+#include <ATen/native/Resize.h>
 #include <ATen/native/TensorCompare.h>
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/TensorCompareKernels.cpp:5]: Is this file change needed

**Comment by majing921201** [comment]
add 'mode' to _xpu_computation_op_list

**Review by xytintel — APPROVED** [review]

**Pull request merged**